File: frontend/demo/component/inputfields/input-field-required.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/date-picker';
import '@vaadin/horizontal-layout';
import '@vaadin/text-field';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('input-field-required')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing">
        <!-- tag::snippet[] -->
        <vaadin-text-field
          label="Name"
          required
          error-message="This field is required"
        ></vaadin-text-field>
        <vaadin-date-picker label="Date of birth"></vaadin-date-picker>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/inputfields/InputFieldRequired.java
package com.vaadin.demo.component.inputfields;

import com.vaadin.flow.component.datepicker.DatePicker;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("input-field-required")
public class InputFieldRequired extends HorizontalLayout {

  public InputFieldRequired() {
    // tag::snippet[]
    TextField textField = new TextField();;
    textField.setLabel("Name");
    textField.setRequiredIndicatorVisible(true);
    textField.setErrorMessage("This field is required");

    DatePicker datePicker = new DatePicker();
    datePicker.setLabel("Date of birth");

    add(textField, datePicker);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<InputFieldRequired> { // hidden-source-line
  } // hidden-source-line
}
